torch/csrc/autograd/custom_function.cpp

```diff
@@ -78,7 +85,42 @@ std::vector<Variable> _wrap_outputs(const std::unordered_set<at::TensorImpl*> &i
   return outputs;
 }
 
+template<class T>
+void CFunction<T>::apply(variable_list&& input_vars) {
+  bool is_executable =  GradMode::is_enabled() && any_variable_requires_grad(input_vars);
+  auto next_edges = collect_next_edges(input_vars);
+
+  std::shared_ptr<CustomFunc<T>> grad_fn = std::make_shared<CustomFunc<T>>();
```

Comment:
You're going to want to use our custom deleter, `deleteFunction`. Example https://github.com/pytorch/pytorch/pull/22983/files#diff-3379b66ac18dc024127ee0a89616eb7eR669 and the sordid backstory at #6873